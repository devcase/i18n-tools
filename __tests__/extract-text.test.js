import extractText from '../src/commons/extract-text'

test("test 1", () => {
    const code = `
        var x = "Este texto precisa ser extraído"
    `
    const results = extractText(code);
    expect(results).toMatchSnapshot()
})

test("test com jsx", () => {
    const code = `
        import React from 'react'
        
        function Component(props) {
            var url = "http://www.ignorar.esta.url.com.br";
            var camelCaseString = "IgnoreThisWord";
            return <div 
                label="Este é um label e precisa ser extraído" 
                className="mb-2 ignorethis"
                max="3">
                Este texto precisa ser extraído
                <span>Não reembolsável</span>
            </div>;
        }
    `
    const results = extractText(code)
    expect(results).toMatchSnapshot()

})


test("test com jsx 2", () => {
    const code = `
        import React from 'react'
        
        function Component(props) {
            return <div 
                className={"mb-2 ignorethis second-time"}
                max="3">
            </div>;
        }
    `
    const results = extractText(code)
    expect(results).toMatchSnapshot()

})

test("teste ignorar jquery", () => {
    const code = `
        import $ from 'jquery'
        
        $('.selectorjquery').hide();
    `
    const results = extractText(code, "teste ignorar jquery")
    expect(results).toMatchSnapshot()

})



test("teste template com jsx", () => {
    const code = "var x = <div className={`room-rate-row row no-gutters ${1==1 ? 'ignorethis' : ''}`}></div>"
    const results = extractText(code, "teste template")
    expect(results).toMatchSnapshot()

})


test("teste template 2", () => {
    const code = "var x = `template precisa ser internacionalizável`"
    const results = extractText(code, "teste template 2")
    expect(results).toMatchSnapshot()

})


test("README", () => {
    const code = `import React from 'react'

    export default function (props) {
        var value = {
            "label": "Incluir este texto",
            text: "Este texto também, mas sem as exclamações!!!!",
            "constantName": "Ignore this text",
            "error": "i18n:Erro de entrada"
        };
        return <div className="my-ignored-classname">
            <h1>     Meu texto internacionalizável!!!!</h1>
            <input type="text" placeholder="Nome do usuário"/>
        </div>
    }
    `
    const results = extractText(code, "README.md")
    expect(results).toMatchSnapshot()
})