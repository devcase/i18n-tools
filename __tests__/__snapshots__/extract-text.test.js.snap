// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`README 1`] = `
Object {
  "hashmap": Object {
    "1105371475": "h1105371475_erro-de-entrada",
    "1350806831": "h1350806831_incluir-este-texto",
    "2161373089": "h2161373089_label",
    "2994568279": "h2994568279_nome-do-usuario",
    "3102868937": "h3102868937_constantname",
    "3122683021": "h3122683021_este-texto-tambem",
    "3251541170": "h3251541170_error",
    "3251814319": "h3251814319_meu-texto-internacionalizavel",
    "3375576029": "h3375576029_text",
    "3796890060": "h3796890060_ignore-this-text",
    "885157645": "h0885157645_react",
    "922874748": "h0922874748_my-ignored-classname",
  },
  "ignored": Object {
    "h0885157645_react": "react",
    "h0922874748_my-ignored-classname": "my-ignored-classname",
    "h2161373089_label": "label",
    "h3102868937_constantname": "constantName",
    "h3251541170_error": "error",
    "h3375576029_text": "text",
    "h3796890060_ignore-this-text": "Ignore this text",
  },
  "strings": Object {
    "h1105371475_erro-de-entrada": "Erro de entrada",
    "h1350806831_incluir-este-texto": "Incluir este texto",
    "h2994568279_nome-do-usuario": "Nome do usuário",
    "h3122683021_este-texto-tambem": "Este texto também",
    "h3251814319_meu-texto-internacionalizavel": "Meu texto internacionalizável",
  },
}
`;

exports[`test 1 1`] = `
Object {
  "hashmap": Object {
    "2951570100": "h2951570100_este-texto-precisa-ser-extraido",
  },
  "ignored": Object {},
  "strings": Object {
    "h2951570100_este-texto-precisa-ser-extraido": "Este texto precisa ser extraído",
  },
}
`;

exports[`test com jsx 1`] = `
Object {
  "hashmap": Object {
    "1090290174": "h1090290174_ignorethisword",
    "1836160639": "h1836160639_http-www-ignorar-esta-url-com-br",
    "2639710208": "h2639710208_mb-2-ignorethis",
    "2951570100": "h2951570100_este-texto-precisa-ser-extraido",
    "4261676902": "h4261676902_este-e-um-label-e-precisa-ser-extraido",
    "4261757570": "h4261757570_nao-reembolsavel",
    "576345883": "h0576345883_3",
    "885157645": "h0885157645_react",
  },
  "ignored": Object {
    "h0576345883_3": "3",
    "h0885157645_react": "react",
    "h1090290174_ignorethisword": "IgnoreThisWord",
    "h1836160639_http-www-ignorar-esta-url-com-br": "http://www.ignorar.esta.url.com.br",
    "h2639710208_mb-2-ignorethis": "mb-2 ignorethis",
  },
  "strings": Object {
    "h2951570100_este-texto-precisa-ser-extraido": "Este texto precisa ser extraído",
    "h4261676902_este-e-um-label-e-precisa-ser-extraido": "Este é um label e precisa ser extraído",
    "h4261757570_nao-reembolsavel": "Não reembolsável",
  },
}
`;

exports[`test com jsx 2 1`] = `
Object {
  "hashmap": Object {
    "576345883": "h0576345883_3",
    "74761743": "h0074761743_mb-2-ignorethis-second-time",
    "885157645": "h0885157645_react",
  },
  "ignored": Object {
    "h0074761743_mb-2-ignorethis-second-time": "mb-2 ignorethis second-time",
    "h0576345883_3": "3",
    "h0885157645_react": "react",
  },
  "strings": Object {},
}
`;

exports[`teste ignorar jquery 1`] = `
Object {
  "hashmap": Object {
    "1997815419": "h1997815419_selectorjquery",
    "2383043738": "h2383043738_jquery",
  },
  "ignored": Object {
    "h1997815419_selectorjquery": ".selectorjquery",
    "h2383043738_jquery": "jquery",
  },
  "strings": Object {},
}
`;

exports[`teste template 2 1`] = `
Object {
  "hashmap": Object {},
  "ignored": Object {},
  "strings": Object {},
}
`;

exports[`teste template com jsx 1`] = `
Object {
  "hashmap": Object {
    "3448505072": "h3448505072_ignorethis",
  },
  "ignored": Object {
    "h3448505072_ignorethis": "ignorethis",
  },
  "strings": Object {},
}
`;
